src/usePreviousDistinct.ts

```diff
@@ -9,7 +9,7 @@ export default function usePreviousDistinct<T>(
   value: T,
   compare: Predicate<T> = strictEquals
 ): T | undefined {
-  const prevRef = useRef<T>();
+  const prevRef = useRef<T | undefined>(undefined);
```

Comment:
The useRef has 1 required parameter now, but there are still the helper overloads for undefined or null, so you don't need to set those explicitly.
```suggestion
  const prevRef = useRef<T>(undefined);
```

See https://github.com/DefinitelyTyped/DefinitelyTyped/blob/8c2f6c1e8c0cdea479233ca90b6f50e3cb545fc0/types/react/index.d.ts#L1746